### DISCUSSION TITLE
Trying to do a discord AI chatbot

### DISCUSSION THREAD

**Discussion by rbustillo** [Rapptz/discord.py#7128] [Q&A]
so I made a chatbot that worked perfectly on the console, but when I tried to insert the discord messages part it doesn't work.
I dont know if someone can help me

**Comment by wasi-master** [comment]
> so I made a chatbot that worked perfectly on the console, but when I tried to insert the discord messages part it doesn't work.
> I dont know if someone can help me
> 
> 

1. Send code
2. What do you mean by doesn't work?
3. What do you mean by inserting the discord messages part
> I don’t know if someone can help me

If you give proper info, only then someone can help you

Edit: Figured out what you mean by the discord messages part

**Reply by bandwiches** [reply]
```python
@client.event
async def on_message(message):
    ...
    message = input("")
    ints = predict_class(message)
    ...
```
You just overwrote the message with a blank string and then passed that to your custom `predict_class()` function.

You already have a Message object via `async def on_message(message)` which maps to [Message](https://discordpy.readthedocs.io/en/stable/api.html?highlight=on_message#discord.Message).

Try this instead...
```python
@client.event
async def on_message(message):
    ...
    # message = input("")
    ints = predict_class(message.content)
    ...
```

**Reply by bandwiches** [reply]
Also, you just shared your bot token with the whole world...twice.  You should probably request a new one.

You may have actually shared even more than your bot token...ouch, big whoopsie.

**Reply by rbustillo** [reply]
> Also, you just shared your bot token with the whole world...twice. You should probably request a new one.

hahahah yeah I know.

I already regenerated it

**Reply by rbustillo** [reply]
> ```python
> @client.event
> async def on_message(message):
>     ...
>     message = input("")
>     ints = predict_class(message)
>     ...
> ```
> 
> You just overwrote the message with a blank string and then passed that to your custom `predict_class()` function.
> 
> You already have a Message object via `async def on_message(message)` which maps to [Message](https://discordpy.readthedocs.io/en/stable/api.html?highlight=on_message#discord.Message).
> 
> Try this instead...
> 
> ```python
> @client.event
> async def on_message(message):
>     ...
>     # message = input("")
>     ints = predict_class(message.content)
>     ...
> ```

@client.event
async def on_message(message):
    if message.author == client.user:
        return

    # message = input("")
    ints = predict_class(message.content)
    res = get_response(ints, intents)
    await message.channel.send(res) 

Is this good?

**Reply by rbustillo** [reply]
> ```python
> @client.event
> async def on_message(message):
>     ...
>     message = input("")
>     ints = predict_class(message)
>     ...
> ```
> 
> You just overwrote the message with a blank string and then passed that to your custom `predict_class()` function.
> 
> You already have a Message object via `async def on_message(message)` which maps to [Message](https://discordpy.readthedocs.io/en/stable/api.html?highlight=on_message#discord.Message).
> 
> Try this instead...
> 
> ```python
> @client.event
> async def on_message(message):
>     ...
>     # message = input("")
>     ints = predict_class(message.content)
>     ...
> ```

It worked 

thank you

**Reply by rbustillo** [reply]
> Also, you just shared your bot token with the whole world...twice. You should probably request a new one.
> 
> You may have actually shared even more than your bot token...ouch, big whoopsie.

what else??

**Reply by bandwiches** [reply]
Good is relative, but yeah.

**Reply by bandwiches** [reply]
> > Also, you just shared your bot token with the whole world...twice. You should probably request a new one.
> > You may have actually shared even more than your bot token...ouch, big whoopsie.
> 
> what else??

Nothing, actually. I was thinking you had some keys in there, but I had the wrong perspective.  I didn't go through everything.
